iceoryx_posh/test/moduletests/test_roudi_cmd_line_parser_config_file_option.cpp

```diff
@@ -97,10 +103,65 @@ TEST_F(CmdLineParserConfigFileOption_test, HelpLongOptionLeadsProgrammNotRunning
     args[0] = &appName[0];
     args[1] = &option[0];
 
-    iox::config::CmdLineParserConfigFileOption sut;
+    CmdLineParserConfigFileOption sut;
     auto result = sut.parse(numberOfArgs, args);
 
-    EXPECT_THAT(result.has_error(), Eq(false));
-    EXPECT_THAT(result.value().run, Eq(false));
+    EXPECT_FALSE(result.has_error());
+    EXPECT_FALSE(result.value().run);
 }
+
+TEST_F(CmdLineParserConfigFileOption_test, WrongOptionLeadsUnkownOptionResult)
+{
+    constexpr uint8_t numberOfArgs{2U};
+    char* args[numberOfArgs];
+    char appName[] = "./foo";
+    char option[] = "--unknown";
+    args[0] = &appName[0];
+    args[1] = &option[0];
+
+    CmdLineParserConfigFileOption sut;
+    auto result = sut.parse(numberOfArgs, args);
+
+    EXPECT_TRUE(result.has_error());
+    EXPECT_THAT(result.get_error(), Eq(CmdLineParserResult::UNKNOWN_OPTION_USED));
+}
+
+TEST_F(CmdLineParserConfigFileOption_test, UnknownOptionLeadsCallingCmdLineParserParseReturningNoError)
```

Comment:
I know that the Parser currently behaves in this way but I would way this is a bug in the parser. If you provide an unknown option it should state that with an error! Could you adjust the parser please so that it fits that requirement.

Reply:
Unfortunately, I cannot update the production code. Another issue should be created for that :)

Reply:
I completely understand your situation in which you are in!

But it makes all our lives harder. The basic idea of unit testing is to verify the behavior and if it is not fitting or you discover and easy to fix bug you fix it. Otherwise one would have a lot of overhead by tracking the issue and generate multiple task where one quick fix would suffice.

To be clear I do not think that this is your fault at all! 

Someone else is responsible for this silly bureaucratic overhead and I hope they read this!

Reply:
I will see what I can do.

Reply:
I think it's a misunderstanding. `UnknownOption` in this tests means unknown for the `CmdLineParserConfigFileOption`. However, every unknown option  in `CmdLineParserConfigFileOption` will be forwarded to `CmdLineParser`.
Tests will be refined to make that clearer!

Reply:
Looks like it's done.